src/PowerFlowData.jl

```diff
@@ -51,23 +86,65 @@ flows and angles, as well as these ones.
 struct PowerFlowData{
     M <: PNM.PowerNetworkMatrix,
     N <: Union{PNM.PowerNetworkMatrix, Nothing},
-}
+} <: PowerFlowContainer
     bus_lookup::Dict{Int, Int}
     branch_lookup::Dict{String, Int}
     bus_activepower_injection::Matrix{Float64}
     bus_reactivepower_injection::Matrix{Float64}
     bus_activepower_withdrawals::Matrix{Float64}
     bus_reactivepower_withdrawals::Matrix{Float64}
-    bus_reactivepower_bounds::Vector{Vector{Float64}}
-    bus_type::Vector{PSY.ACBusTypes}
+    bus_reactivepower_bounds::Matrix{Vector{Float64}}
+    bus_type::Matrix{PSY.ACBusTypes}
+    branch_type::Vector{DataType}
     bus_magnitude::Matrix{Float64}
     bus_angles::Matrix{Float64}
-    branch_flow_values::Matrix{Float64}
+    branch_activepower_flow_from_to::Matrix{Float64}
+    branch_reactivepower_flow_from_to::Matrix{Float64}
+    branch_activepower_flow_to_from::Matrix{Float64}
+    branch_reactivepower_flow_to_from::Matrix{Float64}
     timestep_map::Dict{Int, String}
     valid_ix::Vector{Int}
     power_network_matrix::M
     aux_network_matrix::N
     neighbors::Vector{Set{Int}}
+    converged::Vector{Bool}
+    solver_data::Vector{SolverData}
+end
+
+get_bus_lookup(pfd::PowerFlowData) = pfd.bus_lookup
+get_branch_lookup(pfd::PowerFlowData) = pfd.branch_lookup
+get_bus_activepower_injection(pfd::PowerFlowData) = pfd.bus_activepower_injection
+get_bus_reactivepower_injection(pfd::PowerFlowData) = pfd.bus_reactivepower_injection
+get_bus_activepower_withdrawals(pfd::PowerFlowData) = pfd.bus_activepower_withdrawals
+get_bus_reactivepower_withdrawals(pfd::PowerFlowData) = pfd.bus_reactivepower_withdrawals
+get_bus_reactivepower_bounds(pfd::PowerFlowData) = pfd.bus_reactivepower_bounds
+get_bus_type(pfd::PowerFlowData) = pfd.bus_type
+get_branch_type(pfd::PowerFlowData) = pfd.branch_type
+get_bus_magnitude(pfd::PowerFlowData) = pfd.bus_magnitude
+get_bus_angles(pfd::PowerFlowData) = pfd.bus_angles
+get_branch_activepower_flow_from_to(pfd::PowerFlowData) =
+    pfd.branch_activepower_flow_from_to
+get_branch_reactivepower_flow_from_to(pfd::PowerFlowData) =
+    pfd.branch_reactivepower_flow_from_to
+get_branch_activepower_flow_to_from(pfd::PowerFlowData) =
+    pfd.branch_activepower_flow_to_from
+get_branch_reactivepower_flow_to_from(pfd::PowerFlowData) =
+    pfd.branch_reactivepower_flow_to_from
+get_timestep_map(pfd::PowerFlowData) = pfd.timestep_map
+get_valid_ix(pfd::PowerFlowData) = pfd.valid_ix
+get_power_network_matrix(pfd::PowerFlowData) = pfd.power_network_matrix
+get_aux_network_matrix(pfd::PowerFlowData) = pfd.aux_network_matrix
+get_neighbor(pfd::PowerFlowData) = pfd.neighbors
+supports_multi_period(::PowerFlowData) = true
+get_converged(pfd::PowerFlowData) = pfd.converged
+get_solver_data(pfd::PowerFlowData) = pfd.solver_data
+
+function clear_injection_data!(pfd::PowerFlowData)
```

Comment:
@GabrielKS should we have a clear method that clears all fields?

Reply:
This is used in the power flow in the loop to set all bus injections to zero specifically to enable us to loop over generators and `+=`. I don't think that's relevant for any other fields, so I wouldn't say that's a priority.